### PR TITLE
feat(frontend): integrate config settings with API

### PR DESCRIPTION
**Pull request by jorgeepc**
This PR integrates analytics (config) settings with the resources API.

## Changes

- Integrate web UI with resources API
- Consume `analyticsEnabled` flag from API to trigger web events

## Fixes

- #2100 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: web/src/models/Config.model.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import {EResourceType, TListResponse, TResource} from 'types/Settings.types';
+
+export type TRawConfig = {
```
Comment (jorgeepc) [inline web/src/models/Config.model.ts:3]: We should replace these types with the OpenAPI generated types

File: web/src/providers/SettingsValues/SettingsValues.provider.tsx [hunk 2]
```diff
@@ -66,6 +75,17 @@ const DataStoreConfigProvider = ({children}: IProps) => {
     );
   }, [dispatch]);
 
+  // Config
+  const {data: config = Config()} = useGetConfigQuery({});
+
+  useEffect(() => {
```
Comment (xoscar) [inline web/src/providers/SettingsValues/SettingsValues.provider.tsx:81]: I like this, but let's make sure we don't forget this provider and this side effect is the one handling the analytics switch

**Review by xoscar — APPROVED** [review]
Clean job dude! I left a quick non-blocking question 😄

File: web/public/index.html [hunk 3]
```diff
@@ -78,11 +78,9 @@
...
-        analytics.load(window.ENV.measurementId);
-        }}();
-      }
+      !function(){var analytics=window.analytics=window.analytics||[];if(!analytics.initialize)if(analytics.invoked)window.console&&console.error&&console.error("Segment snippet included twice.");else{analytics.invoked=!0;analytics.methods=["trackSubmit","trackClick","trackLink","trackForm","pageview","identify","reset","group","track","ready","alias","debug","page","once","off","on","addSourceMiddleware","addIntegrationMiddleware","setAnonymousId","addDestinationMiddleware"];analytics.factory=function(e){return function(){var t=Array.prototype.slice.call(arguments);t.unshift(e);analytics.push(t);return analytics}};for(var e=0;e<analytics.methods.length;e++){var key=analytics.methods[e];analytics[key]=analytics.factory(key)}analytics.load=function(key,e){var t=document.createElement("script");t.type="text/javascript";t.async=!0;t.src="https://cdn.segment.com/analytics.js/v1/" + key + "/analytics.min.js";var n=document.getElementsByTagName("script")[0];n.parentNode.insertBefore(t,n);analytics._loadOptions=e};analytics._writeKey="X2vwYODI1vb8g6QpzL9OBxuv5vK7dGan";analytics.SNIPPET_VERSION="4.15.3";
```
Comment (xoscar) [inline web/public/index.html:81]: QQ do we know for sure that by just loading the script we are not sending any telemetry data? 

I want to clarify if the user updates the flag but still after a refresh this scripts sends something we don't want/expect

**Review by xoscar — APPROVED** [review]
Great job doing the analytics lib changes dude!

**Pull request merged**